cli/ingress-controller/flags.go

```diff
@@ -289,49 +261,29 @@ func parseFlags() (cliConfig, error) {
 
 	// Kong connection details
 	kongAdminURL := defaultKongAdminURL
-	oldURL := viper.GetString("kong-url")
-	newURL := viper.GetString("kong-admin-url")
-	if oldURL != "" {
-		kongAdminURL = oldURL
-	}
-	if newURL != defaultKongAdminURL {
-		kongAdminURL = newURL
+	flagURL := viper.GetString("kong-admin-url")
+	if flagURL != defaultKongAdminURL {
+		kongAdminURL = flagURL
 	}
 	config.KongAdminURL = kongAdminURL
 
 	config.KongWorkspace = viper.GetString("kong-workspace")
 	config.KongAdminConcurrency = viper.GetInt("kong-admin-concurrency")
 	config.KongAdminFilterTags = viper.GetStringSlice("kong-admin-filter-tag")
 
-	config.KongAdminHeaders = viper.GetStringSlice("admin-header")
-	kongAdminHeaders := viper.GetStringSlice("kong-admin-header")
-	if len(kongAdminHeaders) > 0 {
-		config.KongAdminHeaders = kongAdminHeaders
-	}
+	config.KongAdminHeaders = viper.GetStringSlice("kong-admin-header")
 
 	kongAdminToken := viper.GetString("kong-admin-token")
 	if kongAdminToken != "" {
 		config.KongAdminHeaders = append(config.KongAdminHeaders,
 			"kong-admin-token:"+kongAdminToken)
 	}
 
-	config.KongAdminTLSSkipVerify = viper.GetBool("admin-tls-skip-verify")
-	kongAdminTLSSkipVerify := viper.GetBool("kong-admin-tls-skip-verify")
-	if kongAdminTLSSkipVerify {
-		config.KongAdminTLSSkipVerify = kongAdminTLSSkipVerify
-	}
+	config.KongAdminTLSSkipVerify = viper.GetBool("kong-admin-tls-skip-verify")
 
-	config.KongAdminTLSServerName = viper.GetString("admin-tls-server-name")
-	kongAdminTLSServerName := viper.GetString("kong-admin-tls-server-name")
-	if kongAdminTLSServerName != "" {
-		config.KongAdminTLSServerName = kongAdminTLSServerName
-	}
+	config.KongAdminTLSServerName = viper.GetString("kong-admin-tls-server-name")
 
-	config.KongAdminCACertPath = viper.GetString("admin-ca-cert-file")
-	kongAdminCACertPath := viper.GetString("kong-admin-ca-cert-file")
-	if kongAdminCACertPath != "" {
-		config.KongAdminCACertPath = kongAdminCACertPath
-	}
+	config.KongAdminCACertPath = viper.GetString("kong-admin-ca-cert-file")
 
 	kongAdminCACert := viper.GetString("kong-admin-ca-cert")
 	if kongAdminCACert != "" {
```

Comment:
optional:

Now that we're removing many `if`s from this function, we can do it once and for all. For example:

`config.KongAdminCACert` is guaranteed to be unset here, so the `if` is redundant, and the assignment can be made directly:

```go
config.KongAdminCACert = viper.GetString(...)
```